src/linker/Linker/CustomAttributeSource.cs

```diff
@@ -35,6 +66,8 @@ public IEnumerable<CustomAttribute> GetCustomAttributes (ICustomAttributeProvide
 					yield return customAttribute;
 			}
 
+			EnsureProcessedAttributeXml (provider);
```

Comment:
Do we really need to process attribute XML for CustomAttributeSource, or is it enough to ensure XML is processed for LinkerAttributesInformation?

Reply:
We do need it for CustomAttributeSource, because we don't always go through LinkerAttributesInformation - for example:
https://github.com/mono/linker/blob/c45a25d77dfe278bd8fa675cb998cb0573dcb1c0/src/linker/Linker.Dataflow/FlowAnnotations.cs#L110

Reply:
In theory everything could go through LinkerAttributeInformation - for example to change the FlowAnnotations to that is pretty simple:https://github.com/mono/linker/compare/master...vitek-karas:UseLinkerAttributeInformationForDataFlow (and that contains a change to parameter type which is the messiest part of that).

Unfortunately I don't think we should do this as it comes with degradation:
* The LinkerAttributeInformation will cache the result - but FlowAnnotations will never ask for this attribute on any member more than once (since it has its own cache) - so we would be paying for cache which is never hit again
* It requires us to create the actual attribute instance - which in this case is pure waste as we want just the one value from it and we will store it in a separate data structure right away.

So let's keep it as is for now - maybe later on we could introduce a new API on Annotations which would not cache and call the custom attributes directly - but that's just layering stuff - not very interested in doing that.

The LinkerAttributeInformation is for cases where we will ask on a member potentially multiple times and we don't hold separate cache for it elsewhere.